### PR TITLE
[skip ci] add initial draft of contribution filtering doc

### PR DESCRIPTION
**Pull request by h-m-m**
### Why

The filtering code provides us a nice level of flexibility in adding new features. As a tradeoff, it's now harder to understand

### What

Added a markdown document to describe the key parts of the architecture of how we sort and filter contributions for display in the Vue.js code. The description here should be fairly accurate for both the current code base and the proposed changes in https://github.com/rubyforgood/mutual-aid/pull/546

### How

a cat banging at a keyboard

### Next Steps

I'd like to finish https://github.com/rubyforgood/mutual-aid/pull/546
### Outstanding Questions, Concerns and Other Notes

Please suggest edits!

### Accessibility

Hopefully this makes our code easier to understand
### Security

Security concerns are not discussed in this document

### Pre-Merge Checklist

- [ ] Security & accessibility have been considered
- ❌ Tests have been added, or an explanation has been given why the features cannot be tested
- [ ] Documentation and comments have been added to the codebase where required
- ❌ Entry added to CHANGELOG.md if appropriate
- [ ] Outstanding questions and concerns have been resolved
- [ ] Any next steps have been turned into Issues or Discussions as appropriate

### CODE REVIEW

File: doc/structure-of-contribution-filters.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+* passing any user changes to the filters back to a Rails endpoint to receive an updated list of Contributions
+
+### Rails structure
+
+The ContributionsController relies on
+* a list of classes that act as types of filters
+* a BrowseFilter class that knows how to turn items selected from the types of filters and use them to query and filter items from the Contributions class
+* a ContributionBlueprint class that 
+* * turns the filtered list of contributions into the format that the Browse page accepts
+* * accepts some options that allow us to conditionally change what shows up (we expect this to help with user authorization features or system configuration features)
```
Comment (exbinary) [inline doc/structure-of-contribution-filters.md:37]: Did you want these two points to be indented under the one above?

**Review by exbinary — APPROVED** [review]
Looks great! 💥

File: doc/structure-of-contribution-filters.md (hunk 1 above)
Comment (h-m-m) [reply doc/structure-of-contribution-filters.md:37]: Yes, I think so

**Review by maebeale — APPROVED** [review]
Thanks for writing this up! :)

**Pull request merged**